libraries/c_sdk/standard/https/src/iot_https_client.c

```diff
@@ -3267,6 +3274,10 @@ IotHttpsReturnCode_t IotHttpsClient_ReadHeader( IotHttpsResponseHandle_t respHan
     IotHttpsResponseParserState_t savedParserState = PARSER_STATE_NONE;
     size_t numParsed = 0;
 
+    /* Disable -Wunused-but-set-variable for local variables used for logging. */
+    ( void ) numParsed;
+    ( void ) pHttpParserErrorDescription;
+
```

Comment:
Same here. General guidance is that using this technique to ignore unused variables should only be done for unused function parameters, and when that is the case there must be strong motivation why the function can meet it's interface contract without using all the parameters.

Reply:
These variables are used in the logging macros, but warnings for them are generated only when logging is disabled. This PR suppresses those warnings for builds with logging disabled.